hw_diag/utilities/gcs_shipper.py

```diff
@@ -42,7 +43,15 @@ def upload_diagnostics(diagnostics, ship):
     upload_url = '%s&name=%s' % (URL, file_name)
     headers = {'Content-Type': 'application/json'}
     diagnostics = add_timestamp_to_diagnostics(diagnostics)
-    content = json.dumps(diagnostics)
+
+    try:
+        # validate and filter diagnostic data as per big query model
+        validated_data = DiagnosticDataModel(**diagnostics)
+    except Exception as err:
+        log.error(f'Failed to enforce biquery data model on diagnostics dict {err}')
```

Comment:
dilemma of error vs warning. this shouldn't fail. But shipping is set as cron for 0th minute of every hour. Sometimes that can be too early to have all values reliably populated and thus it will fail.

Reply:
Is it possible to detect the different error modes by catching different exceptions?

Reply:
It is possible to differentiate between exceptions. Pydantic will mostly raise only validationerror and str(e) is recommended way of getting the useful information for the user. I didn't see any reason to do any special handling for it.